core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4454,20 +4434,21 @@ public void doResources(StaplerRequest req, StaplerResponse rsp) throws IOExcept
      * Checks if container uses UTF-8 to decode URLs. See
      * http://wiki.jenkins-ci.org/display/JENKINS/Tomcat#Tomcat-i18n
      */
+    @Restricted(NoExternalUse.class)
+    @RestrictedSince("since TODO")
+    @Deprecated
     public FormValidation doCheckURIEncoding(StaplerRequest request) throws IOException {
-        // expected is non-ASCII String
-        final String expected = "\u57f7\u4e8b";
-        final String value = fixEmpty(request.getParameter("value"));
-        if (!expected.equals(value))
-            return FormValidation.warningWithMarkup(Messages.Hudson_NotUsesUTF8ToDecodeURL());
-        return FormValidation.ok();
+        return ExtensionList.lookup(URICheckEncodingMonitor.class).get(0).doCheckURIEncoding(request);
```

Comment:
> It would be more relevant to return FormValidation error in such case than to blow u the UI

These deprecated methods aren't used anymore. I tried to be nice here instead of just returning `false` or something similar…